### PR TITLE
Don't require source methods by default.

### PR DESCRIPTION
**Pull request by smimram**
By subtyping, it is not nececessary to require the methods for source arguments.

With this patch the type of `amplify` changes from
```
(?id : string, ?override : string?, {float},
 source(audio='b, video='c, midi='d)
 .{
   time : () -> float,
   shutdown : () -> unit,
   fallible : () -> bool,
   skip : () -> unit,
   seek : (float) -> float,
   is_up : () -> bool,
   remaining : () -> float,
   on_track : ((([string * string]) -> unit)) -> unit,
   on_shutdown : ((() -> unit)) -> unit,
   on_metadata : ((([string * string]) -> unit)) -> unit,
   is_ready : () -> bool,
   id : (() -> string)
 }) -> source(audio='b, video='c, midi='d)
.{
  time : () -> float,
  shutdown : () -> unit,
  fallible : () -> bool,
  skip : () -> unit,
  seek : (float) -> float,
  is_up : () -> bool,
  remaining : () -> float,
  on_track : ((([string * string]) -> unit)) -> unit,
  on_shutdown : ((() -> unit)) -> unit,
  on_metadata : ((([string * string]) -> unit)) -> unit,
  is_ready : () -> bool,
  id : (() -> string)}
```
to
```
(?id : string, ?override : string?, {float},
 source(audio='b, video='c, midi='d)) -> source(audio='b,
video='c, midi='d)
.{
  time : () -> float,
  shutdown : () -> unit,
  fallible : () -> bool,
  skip : () -> unit,
  seek : (float) -> float,
  is_up : () -> bool,
  remaining : () -> float,
  on_track : ((([string * string]) -> unit)) -> unit,
  on_shutdown : ((() -> unit)) -> unit,
  on_metadata : ((([string * string]) -> unit)) -> unit,
  is_ready : () -> bool,
  id : (() -> string)}
```
which is arguably more readable...

### CODE REVIEW

**Comment by toots** [comment]
I've been having tons of issues with that, in particular with source subtyping: `(s:source(...))` not working anymore. I'm super down for it. Let me have another pass at it tho!

**Comment by smimram** [comment]
Good. Note that I have added `source_methods` in order to cast a source to a source with the usual methods.

File: src/lang/lang_types.ml [hunk 1]
```diff
@@ -967,7 +964,7 @@ let rec ( <: ) a b =
               let (o, lbl, t'), l2' = get_param [] l2 in
               (* Check on-the-fly that the types match. *)
               begin
-                try t <: t'
```
Comment (toots) [inline src/lang/lang_types.ml:970]: This seems unrelated to the PR?

**Review by toots — COMMENTED** [review]
PR looks good. Are you sure about that one `lang_types` change?

File: src/lang/lang_types.ml (hunk 1 above)
Comment (smimram) [reply src/lang/lang_types.ml:970]: This one is completely related to the PR: the variance was wrong, and it seems that this is the first time that we are really using subtyping for function arguments (to discard methods)...

**Review by toots — APPROVED** [review]
Looks great, thanks!
